projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -0,0 +1,261 @@
+	 *
+	 * @return void
+	 */
+	public function wpcom_enqueue_subscribe_modal_assets() {
+		if ( $this->is_front_end_single_post() ) {
+			wp_enqueue_style( 'subscribe-modal-css', plugins_url( 'subscribe-modal.css', __FILE__ ), array(), JETPACK__VERSION );
+			wp_enqueue_script( 'subscribe-modal-js', plugins_url( 'subscribe-modal.js', __FILE__ ), array( 'wp-components' ), JETPACK__VERSION, true );
+		}
+	}
+
+	/**
+	 * Returns true if we should enable Newsletter content.
+	 * This is currently limited to lettre theme or newsletter sites.
+	 * We could open it to all themes or site intents.
+	 *
+	 * @return bool
+	 */
+	public function should_enable_subscriber_modal() {
+		$is_lettre          = get_option( 'stylesheet' ) === 'lettre';
+		$is_newsletter_site = get_option( 'site_intent' ) === 'newsletter';
+		$is_modal_enabled   = get_option( 'sm_enabled' ) || false;
+		$is_fse_theme       = wp_is_block_theme();
+
+		$should_enable_subscriber_modal = ( $is_lettre || $is_newsletter_site ) && $is_fse_theme && $is_modal_enabled;
+
+		/**
+		* Allows force-enabling or disabling the Subscriptions modal.
+		*
+		* @since $$next-version$$
+		*
+		* @param bool $should_enable_subscriber_modal Whether the Subscriptions modal should be enabled.
+		*/
+		return (bool) apply_filters( 'jetpack_subscriptions_modal_force_enabled', $should_enable_subscriber_modal );
+	}
+
+	/**
+	 * Returns true if we we are on frontend of single post.
+	 * Note: Because of how WordPress works, this function will
+	 * only return the correct value after a certain point in the
+	 * WordPress loading process. You cannot, for example, use this
+	 * method in the class contructor method - it will always return false.
+	 *
+	 * @return bool
+	 */
+	public function is_front_end_single_post() {
+		return ! is_admin() && is_single() && 'post' === get_post_type();
+	}
+
+	/**
+	 * Returns the slug for the Subcribe template.
+	 *
+	 * @return string
+	 */
+	public function get_subscribe_template_slug() {
+		return 'subscribe-modal';
+	}
+
+	/**
+	 * Returns the initial content of the Subscribe Modal template.
+	 * This can then be edited by the user.
+	 *
+	 * @return string
+	 */
+	public function get_subscribe_template_content() {
+		return '<!-- wp:group {"style":{"spacing":{"padding":{"top":"50px","bottom":"50px","left":"20px","right":"20px"}}},"layout":{"type":"constrained"}} -->
+		<div class="wp-block-group" style="padding-top:50px;padding-right:20px;padding-bottom:50px;padding-left:20px"><!-- wp:group {"style":{"dimensions":{"minHeight":"0px"},"spacing":{"blockGap":"8px"}},"layout":{"type":"flex","flexWrap":"wrap","justifyContent":"center"}} -->
+		<div class="wp-block-group" style="min-height:0px"><!-- wp:heading {"style":{"typography":{"fontStyle":"normal","fontWeight":"600","fontSize":"26px"},"layout":{"selfStretch":"fit","flexSize":null}}} -->
+		<h2 class="wp-block-heading" style="font-size:26px;font-style:normal;font-weight:600">' . __( 'Disover more from', 'jetpack' ) . '</h2>
```

Comment:
```suggestion
		<h2 class="wp-block-heading" style="font-size:26px;font-style:normal;font-weight:600">' . __( 'Discover more from', 'jetpack' ) . '</h2>
```

Reply:
I should add that this may prove difficult to translate for translators, since the site title is outside of the string. In languages where the site title would come before "from", it will be a problem.
It would be nice to have the site title part of that string instead.